### PR TITLE
move to explicit yml for build

### PR DESCRIPTION
**Pull request by SimonCropp**
#skip-changelog

prep for https://github.com/getsentry/sentry-dotnet/issues/1518

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,10 +1,8 @@
...
 
-$testLogger = if ($env:GITHUB_ACTIONS -eq "true") {"GitHubActions;report-warnings=false"} else {"console"}
-
-dotnet test SentryNoSamples.slnf -c Release -l $testLogger `
+dotnet test SentryNoSamples.slnf -c Release -l "GitHubActions;report-warnings=false" `
     /p:CopyLocalLockFileAssemblies=true
 if ($LASTEXITCODE -ne 0) { exit 1 }
 
-dotnet pack SentryNoSamples.slnf -c Release /p:ContinuousIntegrationBuild=true
+dotnet pack SentryNoSamples.slnf -c Release --no-build /p:ContinuousIntegrationBuild=true
```
Comment (mattjohnsonpint) [inline build.ps1:7]: Do we even need to pack via this script at all now?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -57,9 +50,16 @@ jobs:
...
+          run: dotnet build SentryNoSamples.slnf -c Release /p:CopyLocalLockFileAssemblies=true
+
+        - name: Test
+          run: dotnet test SentryNoSamples.slnf -c Release --no-build -l "GitHubActions;report-warnings=false"
+
+        - name: Pack
+          if: matrix.os == 'windows-2022'
+          run: dotnet pack SentryNoSamples.slnf -c Release --no-build /p:ContinuousIntegrationBuild=true
 
         - name: Publish coverage report
```
Comment (mattjohnsonpint) [inline .github/workflows/build.yml:64]: I thought we were no longer doing code coverage, so should we remove this step?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -57,9 +50,16 @@ jobs:
...
           uses: codecov/codecov-action@v1
```
Comment (mattjohnsonpint) [inline .github/workflows/build.yml:65]: I thought we were no longer doing code coverage, so should we remove this step?

File: build.ps1 (hunk 1 above)
Comment (SimonCropp) [reply build.ps1:7]: i assume if someone is using the script to build, they might also want the nugets for local testing?

File: .github/workflows/build.yml (hunk 3 above)
Comment (SimonCropp) [reply .github/workflows/build.yml:65]: https://github.com/getsentry/sentry-dotnet/pull/1588

File: build.ps1 (hunk 1 above)
Comment (SimonCropp) [reply build.ps1:7]: i will defer to you and @bruno-garcia on this. i am pretty comfortable with calling build/test/pack explicitly on the rare occasions i need to

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**